File: c++/src/semaine/datatypes/xml/FML.cpp
/*
 *  FML.cpp
 *  semaine
 *
 *  Created by Marc Schröder on 06.11.08.
 *  Copyright 2008 DFKI GmbH. All rights reserved.
 *
 */

#include "FML.h"

namespace semaine {
namespace datatypes {
namespace xml {

const std::string FML::version = "0.1";
const std::string FML::namespaceURI = "http://www.mindmakers.org/fml";
	
	// Elements
const std::string FML::E_FML = "fml";
const std::string FML::E_ROOT_TAGNAME = "fml";
const std::string FML::E_PERFORMATIVE = "performative";
const std::string FML::E_THEME = "theme";
const std::string FML::E_RHEME = "rheme";
const std::string FML::E_WORLD = "world";
const std::string FML::E_EMOTION = "emotion";

	// Attributes
const std::string FML::A_ID = "id";
const std::string FML::A_START = "start";
const std::string FML::A_END = "end";
const std::string FML::A_TYPE = "type";

} // namespace xml
} // namespace datatypes
} // namespace semaine

File: java/src/eu/semaine/datatypes/xml/FML.java
/**
 * Copyright (C) 2008 DFKI GmbH. All rights reserved.
 * Use is subject to license terms -- see license.txt.
 */
package eu.semaine.datatypes.xml;

/**
 * @author marc
 *
 */
public class FML 
{
    public static final String version = "0.1";
    public static final String namespaceURI = "http://www.mindmakers.org/projects/FML";

    // Elements
    public static final String E_FML = "fml";
	public static final String ROOT_TAGNAME = E_FML;
	public static final String E_PERFORMATIVE = "performative";
	public static final String E_THEME = "theme";
	public static final String E_RHEME = "rheme";
	public static final String E_WORLD = "world";
	public static final String E_EMOTION = "emotion";
	public static final String E_BACKCHANNEL = "backchannel";

	// Attributes
	public static final String A_ID = "id";
	public static final String A_START = "start";
	public static final String A_END = "end";
	public static final String A_TYPE = "type";


}
